Error: Can't resolve 'vm'

I created a simple [Angular application](https://github.com/esanzgar/cwlts-angular) and imported cwlts/models. When the application is compiled it complains about 'vm' module not present.

It is fixed if I install 'vm' (`npm install vm`).

Maybe you should add the 'vm' module to the list of dependencies in your package.json.

```stacktrace
ERROR in ./node_modules/cwlts/models/helpers/JSExecutor.js
Module not found: Error: Can't resolve 'vm' in '/Users/eduardo/Development/glue/cwlts-angular/node_modules/cwlts/models/helpers'
resolve 'vm' in '/Users/eduardo/Development/glue/cwlts-angular/node_modules/cwlts/models/helpers'
  Parsed request is a module
  using description file: /Users/eduardo/Development/glue/cwlts-angular/node_modules/cwlts/package.json (relative path: ./models/helpers)
    Field 'browser' doesn't contain a valid alias configuration
    resolve as module
      looking for modules in /Users/eduardo/Development/glue/cwlts-angular/
        using description file: /Users/eduardo/Development/glue/cwlts-angular/package.json (relative path: .)
          Field 'browser' doesn't contain a valid alias configuration
          using description file: /Users/eduardo/Development/glue/cwlts-angular/package.json (relative path: ./vm)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              /Users/eduardo/Development/glue/cwlts-angular/vm doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              /Users/eduardo/Development/glue/cwlts-angular/vm.ts doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /Users/eduardo/Development/glue/cwlts-angular/vm.js doesn't exist
            as directory
              /Users/eduardo/Development/glue/cwlts-angular/vm doesn't exist
      /Users/eduardo/Development/glue/cwlts-angular/node_modules/cwlts/models/helpers/node_modules doesn't exist or is not a directory
      /Users/eduardo/Development/glue/cwlts-angular/node_modules/cwlts/models/node_modules doesn't exist or is not a directory
      /Users/eduardo/Development/glue/cwlts-angular/node_modules/node_modules doesn't exist or is not a directory
      /Users/eduardo/Development/glue/node_modules doesn't exist or is not a directory
      /Users/eduardo/Development/node_modules doesn't exist or is not a directory
      /Users/eduardo/node_modules doesn't exist or is not a directory
      /Users/node_modules doesn't exist or is not a directory
      /node_modules doesn't exist or is not a directory
      looking for modules in /Users/eduardo/Development/glue/cwlts-angular/node_modules/cwlts/node_modules
        using description file: /Users/eduardo/Development/glue/cwlts-angular/node_modules/cwlts/package.json (relative path: ./node_modules)
          Field 'browser' doesn't contain a valid alias configuration
      looking for modules in /Users/eduardo/Development/glue/cwlts-angular/node_modules
        using description file: /Users/eduardo/Development/glue/cwlts-angular/package.json (relative path: ./node_modules)
          Field 'browser' doesn't contain a valid alias configuration
          using description file: /Users/eduardo/Development/glue/cwlts-angular/node_modules/cwlts/package.json (relative path: ./node_modules/vm)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
          using description file: /Users/eduardo/Development/glue/cwlts-angular/package.json (relative path: ./node_modules/vm)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              /Users/eduardo/Development/glue/cwlts-angular/node_modules/cwlts/node_modules/vm doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              /Users/eduardo/Development/glue/cwlts-angular/node_modules/vm doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              /Users/eduardo/Development/glue/cwlts-angular/node_modules/cwlts/node_modules/vm.ts doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /Users/eduardo/Development/glue/cwlts-angular/node_modules/vm.ts doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /Users/eduardo/Development/glue/cwlts-angular/node_modules/cwlts/node_modules/vm.js doesn't exist
              /Users/eduardo/Development/glue/cwlts-angular/node_modules/vm.js doesn't exist
            as directory
              /Users/eduardo/Development/glue/cwlts-angular/node_modules/cwlts/node_modules/vm doesn't exist
            as directory
              /Users/eduardo/Development/glue/cwlts-angular/node_modules/vm doesn't exist
[/Users/eduardo/Development/glue/cwlts-angular/vm]
[/Users/eduardo/Development/glue/cwlts-angular/vm.ts]
[/Users/eduardo/Development/glue/cwlts-angular/vm.js]
[/Users/eduardo/Development/glue/cwlts-angular/node_modules/cwlts/models/helpers/node_modules]
[/Users/eduardo/Development/glue/cwlts-angular/node_modules/cwlts/models/node_modules]
[/Users/eduardo/Development/glue/cwlts-angular/node_modules/node_modules]
[/Users/eduardo/Development/glue/node_modules]
[/Users/eduardo/Development/node_modules]
[/Users/eduardo/node_modules]
[/Users/node_modules]
[/node_modules]
[/Users/eduardo/Development/glue/cwlts-angular/node_modules/cwlts/node_modules/vm]
[/Users/eduardo/Development/glue/cwlts-angular/node_modules/vm]
[/Users/eduardo/Development/glue/cwlts-angular/node_modules/cwlts/node_modules/vm.ts]
[/Users/eduardo/Development/glue/cwlts-angular/node_modules/vm.ts]
[/Users/eduardo/Development/glue/cwlts-angular/node_modules/cwlts/node_modules/vm.js]
[/Users/eduardo/Development/glue/cwlts-angular/node_modules/vm.js]
 @ ./node_modules/cwlts/models/helpers/JSExecutor.js 3:9-22
 @ ./node_modules/cwlts/models/helpers/index.js
 @ ./node_modules/cwlts/models/index.js
 @ ./src/app/app.component.ts
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi ./src/main.ts
```

--- a/src/models/d2sb/SBDraft2CommandInputParameterModel.spec.ts
+++ b/src/models/d2sb/SBDraft2CommandInputParameterModel.spec.ts
@@ -2,7 +2,7 @@ import {SBDraft2CommandInputParameterModel} from "./SBDraft2CommandInputParamete
 import {expect} from "chai";
 import {CommandInputParameter} from "../../mappings/d2sb/CommandInputParameter";
 import {ExpressionClass} from "../../mappings/d2sb/Expression";
-import {JSExecutor} from "../helpers/JSExecutor";
+import {JSExecutor} from "../../tests/shared/JSExecutor";
 import {ExpressionEvaluator} from "../helpers/ExpressionEvaluator";
 
 describe("SBDraft2CommandInputParameterModel d2sb", () => {

--- a/src/models/d2sb/SBDraft2CommandLineBindingModel.spec.ts
+++ b/src/models/d2sb/SBDraft2CommandLineBindingModel.spec.ts
@@ -2,7 +2,7 @@ import {expect} from "chai";
 import {CommandLineBinding} from "../../mappings/d2sb/CommandLineBinding";
 import {ExpressionClass} from "../../mappings/d2sb/Expression";
 import {ExpressionEvaluator} from "../helpers/ExpressionEvaluator";
-import {JSExecutor} from "../helpers/JSExecutor";
+import {JSExecutor} from "../../tests/shared/JSExecutor";
 import {SBDraft2CommandLineBindingModel} from "./SBDraft2CommandLineBindingModel";
 
 describe("SBDraft2CommandLineBindingModel d2sb", () => {

--- a/src/models/d2sb/SBDraft2CommandLineToolModel.spec.ts
+++ b/src/models/d2sb/SBDraft2CommandLineToolModel.spec.ts
@@ -10,7 +10,7 @@ import * as BfctoolsAnnotate from "../../tests/apps/bfctools-annotate-sbg";
 import * as BindingTestTool from "../../tests/apps/binding-test-tool";
 import {CommandLineTool} from "../../mappings/d2sb/CommandLineTool";
 import {ExpressionEvaluator} from "../helpers/ExpressionEvaluator";
-import {JSExecutor} from "../helpers/JSExecutor";
+import {JSExecutor} from "../../tests/shared/JSExecutor";
 
 should();
 

--- a/src/models/d2sb/SBDraft2CommandOutputParameterModel.spec.ts
+++ b/src/models/d2sb/SBDraft2CommandOutputParameterModel.spec.ts
@@ -3,7 +3,7 @@ import {CommandOutputBinding} from "../../mappings/d2sb/CommandOutputBinding";
 import {ExpressionClass} from "../../mappings/d2sb/Expression";
 import {testCommandOutputBindingSerialization} from '../../tests/shared/model';
 import {ExpressionEvaluator} from "../helpers/ExpressionEvaluator";
-import {JSExecutor} from "../helpers/JSExecutor";
+import {JSExecutor} from "../../tests/shared/JSExecutor";
 import {SBDraft2CommandLineToolModel} from "./SBDraft2CommandLineToolModel";
 import {SBDraft2CommandOutputBindingModel} from "./SBDraft2CommandOutputBindingModel";
 import {SBDraft2CommandOutputParameterModel} from "./SBDraft2CommandOutputParameterModel";

--- a/src/models/d2sb/SBDraft2ExpressionModel.spec.ts
+++ b/src/models/d2sb/SBDraft2ExpressionModel.spec.ts
@@ -2,7 +2,7 @@ import {expect} from "chai";
 import {SBDraft2ExpressionModel} from "./SBDraft2ExpressionModel";
 import {Expression} from "../../mappings/d2sb/Expression";
 import {ExpressionClass} from "../../mappings/d2sb/Expression";
-import {JSExecutor} from "../helpers/JSExecutor";
+import {JSExecutor} from "../../tests/shared/JSExecutor";
 import {ExpressionEvaluator} from "../helpers/ExpressionEvaluator";
 
 describe("SBDraft2ExpressionModel", () => {

--- a/src/models/helpers/index.ts
+++ b/src/models/helpers/index.ts
@@ -1,6 +1,5 @@
 export * from "./ExpressionEvaluator";
 export * from "./CommandLinePart";
-export * from "./JSExecutor";
 export * from "./Graph";
 export * from "./TypeResolver";
 export * from "./constants";

--- a/src/models/v1.0/V1CommandLineToolModel.spec.ts
+++ b/src/models/v1.0/V1CommandLineToolModel.spec.ts
@@ -3,12 +3,11 @@ import {CommandLineTool} from "../../mappings/v1.0/CommandLineTool";
 import {testNamespaces} from "../../tests/shared/model";
 import {CommandLinePart} from "../helpers/CommandLinePart";
 import {ExpressionEvaluator} from "../helpers/ExpressionEvaluator";
-import {JSExecutor} from "../helpers/JSExecutor";
+import {JSExecutor} from "../../tests/shared/JSExecutor";
 import {V1CommandInputParameterModel} from "./V1CommandInputParameterModel";
 import {V1CommandLineToolModel} from "./V1CommandLineToolModel";
 import {V1CommandOutputParameterModel} from "./V1CommandOutputParameterModel";
 import {V1ExpressionModel} from "./V1ExpressionModel";
-import {CommandLineParsers} from "../helpers/CommandLineParsers";
 
 function runTest(app: CommandLineTool, job: any, expected: CommandLinePart[], done) {
     let model = new V1CommandLineToolModel(app, "document");

--- a/src/tests/shared/JSExecutor.ts
+++ b/src/tests/shared/JSExecutor.ts
@@ -1,4 +1,3 @@
-declare function require(name: string);
 const vm = require('vm');
 
 export class JSExecutor {

